### PR TITLE
Add citation.cff

### PR DESCRIPTION
**Pull request by tmigot**

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  - given-names: Dominique
+    family-names: Orban
+    orcid: 'https://orcid.org/0000-0002-8017-7687'
+    affiliation: >-
+      GERAD and Department of Mathematics
+      and                   Industrial Engineering,
+      Polytechnique       Montréal,             QC, Canada
+    email: dominique.orban@gerad.ca
+  - given-names: Abel Soares
+    family-names: Siqueira
```
Comment (abelsiqueira) [inline CITATION.cff:20]:
```suggestion
  - given-names: Abel
    family-names: Soares Siqueira
```

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    value: 10.5281/zenodo.3700941
+    description: Zenodo
+repository-code: 'https://github.com/JuliaSmoothOptimizers/AmplNLReader.jl'
+keywords:
+  - AMPL
+  - Optimization
+  - Nonlinear programming
+  - Julia
+license: MPL-2.0
+commit: 15165b0086abe9dd05b0ba34b3076ee5e20cb156
```
Comment (abelsiqueira) [inline CITATION.cff:36]:
```suggestion
```
I would remove this. We have a DOI, a version and a date-released, so it is more than sufficient.

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  metadata from this file.
+type: software
+authors:
+  - given-names: Dominique
+    family-names: Orban
+    orcid: 'https://orcid.org/0000-0002-8017-7687'
+    affiliation: >-
+      GERAD and Department of Mathematics
+      and                   Industrial Engineering,
+      Polytechnique       Montréal,             QC, Canada
```
Comment (abelsiqueira) [inline CITATION.cff:17]: Were these extra spaces added automatically by cffinit?

**Review by abelsiqueira — CHANGES_REQUESTED** [review]
I've made some comments, thanks for the work.

File: CITATION.cff (hunk 3 above)
Comment (tmigot) [reply CITATION.cff:17]: Oh no, it is just a copy paste from somewhere else

Comment (tmigot) [inline CITATION.cff:17]:
```suggestion
    affiliation: 'GERAD and Department of Mathematics and Industrial Engineering, Polytechnique Montréal, QC, Canada'
```

**Pull request merged**